src/amms/error.rs

```diff
@@ -1,15 +1,46 @@
+use alloy::{
+    primitives::{Address, FixedBytes},
+    transports::TransportErrorKind,
+};
 use thiserror::Error;
+use uniswap_v3_math::error::UniswapV3MathError;
 
 #[derive(Error, Debug)]
 pub enum AMMError {
     #[error(transparent)]
-    UniswapV3MathError(#[from] uniswap_v3_math::error::UniswapV3MathError),
-    #[error("Liquidity Underflow")]
-    LiquidityUnderflow,
+    TransportError(#[from] alloy_json_rpc::RpcError<TransportErrorKind>),
+    #[error(transparent)]
+    ContractError(#[from] alloy::contract::Error),
+    #[error(transparent)]
+    ABIError(#[from] alloy::dyn_abi::Error),
+    #[error(transparent)]
+    SolTypesError(#[from] alloy::sol_types::Error),
+    #[error(transparent)]
+    UniswapV2Error(#[from] UniswapV2Error),
+    #[error(transparent)]
+    UniswapV3Error(#[from] UniswapV3Error),
+    #[error("Invalid AMM Address")]
+    InvalidAMMAddress(Address),
```

Comment:
I think we can probably remove this error. In the instances that it is used, it is safe to unwrap.